### PR TITLE
Adds support for Math500 accuracy benchmarks

### PR DESCRIPTION
**Pull request by patemotter**
## Key changes:

*Added math500 dataset: The math500 dataset is now supported for accuracy evaluations.

*Integrated lm-evaluation-harness: The Docker image and local run scripts now install `lm-eval` with the necessary dependencies for math evaluations.

*New lm-eval task: A custom lm-eval task for math500 has been added, including a template, task definition, and utility functions.

*Parsing script: A new script has been added to parse the JSON output from lm-eval and extract the relevant accuracy metrics.

## Testing:
Tested locally via the `create_job.sh`, `run_job.sh`, and `report_result.sh` method. 

Note: This code will be refactored or at least cleaned up a bit when support is added for other datasets or tests.

### CODE REVIEW

File: cases/accuracy_jax.csv [hunk 1]
```diff
@@ -1,3 +1,4 @@
 Device,Model,MaxNumSeqs,MaxNumBatchedTokens,TensorParallelSize,MaxModelLen,Dataset,InputLen,OutputLen,ExpectedETEL,NumPrompts
 v6e-8,meta-llama/Llama-3.3-70B-Instruct,128,1024,8,2048,mmlu,1800,128,,1000
 v6e-8,meta-llama/Llama-3.3-70B-Instruct,128,1024,8,2048,mlperf,1024,1024,,1000
+v6e-8,meta-llama/Llama-3.3-70B-Instruct,128,1024,8,2048,math500,1024,1024,,500
```
Comment (QiliangCui) [inline cases/accuracy_jax.csv:4]: nit: add an empty line in the end!

File: scripts/agent/run_bm.sh [hunk 2]
```diff
@@ -2,11 +2,21 @@
 
 set -euo pipefail
 
+# TODO: Move to image building.
+# Ingore the error because in case of using uv, the packages are installed outside this script.
+pip install pandas || true
+pip install datasets || true
+pip install lm-eval[math]==0.4.9.1 || true
```
Comment (QiliangCui) [inline scripts/agent/run_bm.sh:9]: let's double check this: https://github.com/vllm-project/vllm/blob/main/.buildkite/scripts/hardware_ci/run-tpu-v1-test.sh#L64

at least, let's be aware that we are not using exactly the same lm-eval.

https://github.com/vllm-project/vllm/blob/main/.buildkite/scripts/hardware_ci/run-tpu-v1-test.sh#L64

the result maybe different.
